Meta/CMake/get_cxx_includes.py

```diff
@@ -0,0 +1,90 @@
+import os
+import sys
+import subprocess
+
+def get_includes_from_version_output(compiler):
+    try:
+        output = subprocess.check_output([compiler, "-xc++", "/dev/null", "-E", "-Wp,-v"], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print("Error getting includes from compiler version output: " + e.output)
+        sys.exit(1)
+    
+    includes = []
+    in_search_list = False
+    for line in output.splitlines():
+        line = line.strip()
+        if not in_search_list:
+            if line == '#include <...> search starts here:':
+                in_search_list = True
+                continue
+            continue
+        if line == 'End of search list.':
+            break
+        includes.append(line)
+    
+    return includes
+
+
+def get_includes_from_preprocessor_output(compiler, test_file):
+    try:
+        output = subprocess.check_output([compiler, "-E", test_file], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print("Error getting includes from preprocessor output: " + e.output)
+        sys.exit(1)
+    
+    includes = []
+    for line in output.splitlines():
+        if not line.startswith(b"# 1 "):
+            continue
+        include = line[4:]
+        include = include[1:-1] # Remove quotes
+        includes.append(include)
+    
+    return includes
+
+def convert_includes_to_dirs(includes):
+    dirs = []
+    for include in includes:
+        if not os.path.isabs(include):
+            continue
+        include_dir = os.path.dirname(include)
+        if include_dir.endswith(b"/bits"):
+            include_dir = os.path.dirname(include_dir)
+
+        while not os.path.isdir(include_dir):
+            include_dir = os.path.dirname(include_dir)
+
+        dirs.append(include_dir)
+    
+    return dirs
+
+
+if __name__ == "__main__":
+    compiler = None
```

Comment:
nit: put contents here in `def main()` so that variables used in it are function-scoped, not global

(so you won't accidentally read from other functions when you typo a var name there etc)